src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplay.cs

```diff
@@ -107,6 +157,8 @@ public static class SymbolDisplay
                 throw new ArgumentNullException(nameof(symbol));
             }
 
+            Debug.Assert(!topLevelNullabilityOpt.HasValue || symbol is ITypeSymbol);
```

Comment:
Why a debug assert? Either you're willingly throwing it away (and that's OK) or this should be an exception?